packages/app-mobile/components/screens/encryption-config.tsx

```diff
@@ -226,16 +233,19 @@ const EncryptionConfigScreen = (props: Props) => {
 		}
 	};
 
-
-
-	for (let i = 0; i < props.masterKeys.length; i++) {
+	for (let i = 0; i < props.masterKeys.filter(mk => masterKeyEnabled(mk)).length; i++) {
 		const mk = props.masterKeys[i];
-		mkComps.push(renderMasterKey(i + 1, mk));
+		mkComps.push(renderMasterKey(mk));
 
 		const idx = nonExistingMasterKeyIds.indexOf(mk.id);
 		if (idx >= 0) nonExistingMasterKeyIds.splice(idx, 1);
 	}
 
+	for (let i = 0; i < props.masterKeys.filter(mk => !masterKeyEnabled(mk)).length; i++) {
```

Comment:
Yes that logic is not right - it's going to include random keys, not just disabled one.

Should probably be `for (const disabledKey of props.masterKeys.filter(mk => !masterKeyEnabled(mk)))` or something like this.

@pedr